src/Rector/AbstractPHPUnitRector.php

```diff
@@ -2,21 +2,31 @@
 
 namespace Rector\Rector;
 
-use Nette\Utils\Strings;
 use PhpParser\Node;
-use Rector\Node\Attribute;
+use PHPUnit\Framework\TestCase;
+use Rector\NodeTypeResolver\NodeTypeResolver;
 
 abstract class AbstractPHPUnitRector extends AbstractRector
 {
     /**
-     * Check if the file is a PHPUnit TestCase. By default, it should end with "Test", as it is the standard.
+     * @var NodeTypeResolver
+     */
+    private $nodeTypeResolver;
+
+    /**
+     * Nasty magic, unable to do that in config autowire _instanceof calls.
      *
-     * @see https://phpunit.de/getting-started-with-phpunit.html
+     * @required
      */
+    public function setNodeTypeResolver(NodeTypeResolver $nodeTypeResolver): void
+    {
+        $this->nodeTypeResolver = $nodeTypeResolver;
+    }
+
     protected function isInTestClass(Node $node): bool
     {
-        $className = (string) $node->getAttribute(Attribute::CLASS_NAME);
+        $nodeResolved = $this->nodeTypeResolver->resolve($node);
 
-        return Strings::endsWith($className, 'Test');
+        return ! array_intersect([TestCase::class, 'PHPUnit_Framework_TestCase'], $nodeResolved);
```

Comment:
What happens, if sb installs a Rector but doesn't have PHPUnit with `TestCase` class installed?

Reply:
I didn’t get it :(

Without `TestCase` isn’t possible to execute tests, is?

Reply:
Well, somebody can use `PHPUnit_Framework_TestCase`, right?

Reply:
We are checking it, aren't we? :thinking: 
```php
return ! array_intersect([TestCase::class, 'PHPUnit_Framework_TestCase'], $nodeResolved);
```

Reply:
Yes, but that depends on `TestCase` to exist. And it does in our code.

Sb might download Rector and have only `PHPUnit_Framework_TestCase` in their code.
Soe they `TestCase` will end un as unexisting class.

That's why it should string here.

Reply:
So I'll ignore this file from our CS. I used `::class` here because CS complained about it